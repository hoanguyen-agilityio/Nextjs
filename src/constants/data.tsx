const INPUT_FIELD_DATA = [
  {
    type: 'text',
    label: 'Name *',
    message:
      'Give your product a short and clear name. 30 characters is recommended.',
    className: 'mt-2',
    size: 'xl' as const,
    required: true,
    pattern: /^[a-zA-Z\s]+$/,
    minLength: 3,
    maxLength: 100,
  },
  {
    type: 'text',
    label: 'Description',
    message: 'Give your product a short and clear description.',
    classWrapper: 'mt-3xl',
    className: 'mt-2',
    size: 'xl' as const,
    required: false,
  },
  {
    type: 'number',
    label: 'Price *',
    placeholder: '0.00',
    message: 'Give your product a final price.',
    size: 'xxl' as const,
    startContent: (
      <div className="pointer-events-none flex items-center">
        <span className="text-xsm">$</span>
      </div>
    ),
    classWrapper: 'mt-3xl',
    className: 'mt-2',
    required: true,
    pattern: /^[0-9]+(\.[0-9]{1,2})?$/,
  },
];

const KEY_SWITCH = [
  {
    title: 'Generate License Keys',
    describe: 'Issue each customer a unique license key after purchase',
    classWrapper: 'mt-9',
    ariaLabel: 'Generate License Keys',
  },
  {
    title: 'Redirect customers after purchase?',
    describe: 'Redirect customers to a custom URL after making a purchase',
    classWrapper: 'mt-[60px]',
    ariaLabel: 'Redirect customers after purchase?',
  },
  {
    title: 'Display product on shop page?',
    describe: 'Show this product on your shop page',
    classWrapper: 'mt-[60px]',
    defaultSelected: true,
    ariaLabel: 'Display product on shop page?',
  },
];

const CUSTOM_INPUT_FIELDS = [
  {
    label: 'Download Button',
    message: 'Customize the button text in the email receipt.',
    size: 'xl' as const,
    classWrapper: 'mt-[60px]',
    className: 'mt-2',
  },
  {
    label: 'Product Link',
    message: 'Customize the button link in the email receipt.',
    size: '3xl' as const,
    classWrapper: 'mt-[34px]',
    className: 'mt-2',
    startContent: (
      <div className="bg-translucent-100 dark:bg-translucent-700 w-24 h-full rounded-l-lg absolute left-0 flex items-center justify-center">
        <p className="text-xsm text-grayBlue-400">https://</p>
      </div>
    ),
  },
  {
    label: 'Personal Note',
    message: 'Add a personalized thank you note to the end of email receipts.',
    size: 'xl' as const,
    classWrapper: 'mt-[34px]',
    className: 'mt-2',
    placeholder: 'Thank you for purchasing my product!',
  },
];

export { INPUT_FIELD_DATA, KEY_SWITCH, CUSTOM_INPUT_FIELDS };
