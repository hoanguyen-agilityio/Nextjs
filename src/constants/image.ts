const IMAGE = {
  DEFAULT: 'https://picsum.photos/200/300',
  BLUR: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSnOyW9pu8TBhOqw5tUQh_5A6lrIYfmpu4c3GkV7t6pcfu7oTILl9FAm2nX8n45uLHdSpk&usqp=CAU',
};

export { IMAGE };
