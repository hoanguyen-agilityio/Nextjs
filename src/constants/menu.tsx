import dynamic from 'next/dynamic';
import { ROUTERS } from './router';

const Dashboard = dynamic(() => import('@/icons/Dashboard'));
const Inbox = dynamic(() => import('@/icons/Inbox'));
const Wallet = dynamic(() => import('@/icons/Wallet'));
const NewsFeed = dynamic(() => import('@/icons/NewsFeed'));
const Analytics = dynamic(() => import('@/icons/Analytics'));
const Market = dynamic(() => import('@/icons/Market'));
const Archive = dynamic(() => import('@/icons/Archive'));
const Profile = dynamic(() => import('@/icons/Profile'));
const Settings = dynamic(() => import('@/icons/Settings'));

export const MENU = [
  {
    icon: <Dashboard width="30px" height="30px" />,
    label: 'Dashboard',
    href: ROUTERS.HOME,
  },
  {
    icon: <Inbox width="30px" height="30px" />,
    label: 'Inbox',
    href: ROUTERS.INBOX,
  },
  {
    icon: <Wallet width="30px" height="30px" />,
    label: 'Wallet',
    href: ROUTERS.WALLET,
  },
  {
    icon: <NewsFeed width="30px" height="30px" />,
    label: 'NewsFeed',
    href: ROUTERS.NEWS_FEED,
  },
  {
    icon: <Analytics width="30px" height="30px" />,
    label: 'Analytics',
    href: ROUTERS.ANALYTICS,
  },
  {
    icon: <Market width="30px" height="30px" />,
    label: 'Market',
    href: ROUTERS.MARKET,
  },
  {
    icon: <Archive width="30px" height="30px" />,
    label: 'Archive',
    href: ROUTERS.ARCHIVE,
  },
  {
    icon: <Profile width="30px" height="30px" />,
    label: 'Profile',
    className: 'mt-[100px]',
    href: ROUTERS.PROFILE,
  },
  {
    icon: <Settings width="30px" height="30px" />,
    label: 'Settings',
    href: ROUTERS.SETTINGS,
  },
];

export const MENU_NO_AUTH = [
  {
    label: 'Overview',
    href: ROUTERS.EMPTY,
  },
  {
    label: 'Projects',
    href: ROUTERS.EMPTY,
  },
  {
    label: 'Tasks',
    href: ROUTERS.EMPTY,
  },
  {
    label: 'Reports',
    href: ROUTERS.EMPTY,
  },
  {
    label: 'Statements',
    href: ROUTERS.EMPTY,
  },
];
