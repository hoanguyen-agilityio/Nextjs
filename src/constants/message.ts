const EMPTY_TEXT = '';
const MESSAGE = {
  ERROR_URL: 'PRODUCT_URL is undefined',
  ERROR_GET_DATA_PRODUCT: 'Error fetching product data: ',
  ERROR_GET_DATA_OVERVIEW: 'Error fetching overview data:',
  DEFAULT:
    'We apologize for the inconvenience. The page you are looking for might have been removed or is temporarily unavailable.',
  NAME_REQUIRED: 'Name is required',
  PRICE_REQUIRED: 'Price is required',
  DESCRIPTION_REQUIRED: 'Description is required',
  VIEW_REQUIRED: 'View is required',
  SALES_REQUIRED: 'Sales is required',
  CONVERSION_REQUIRED: 'Conversion is required',
  VALID_NUMBER: 'Please enter a valid number.',
  VALID_PERCENT: 'Please enter a valid percentage.',
  ERROR_GET_ACCOUNT: 'An error occurred while retrieving account data.',
  USERNAME_REQUIRED: 'Username is required',
  PASSWORD_REQUIRED: 'Password is required',
  INVALID_USERNAME:
    'Username includes uppercase and lowercase letters and numbers',
  INVALID_PASSWORD:
    'Password must include uppercase and lowercase letters, numbers and special characters',
  DUPLICATE_NAME: 'Name already exists',
  ADD_SUCCESSFULLY: 'Product added successfully',
  EDIT_SUCCESSFULLY: 'Product edit successful',
  DELETE_SUCCESSFULLY: 'Product deleted successfully',
  LOGIN_SUCCESSFULLY: 'Login successful',
  ERROR_ADD_DATA: 'Error adding product:',
  ERROR_EDIT_DATA: 'Error edit product:',
  ERROR_DELETE_DATA: 'Failed to delete product:',
  ERROR_UNKNOWN: 'Unknown error occurred',
  INVALID_ACCOUNT: 'Incorrect username or password',
  IMAGE_OVERLOAD: 'You can only upload up to 10 images.',
  FILE_OVERLOAD: 'File size exceeds the maximum limit of 5GB.',
  MAX_NAME: 'Name cannot exceed 30 characters',
  MAX_STATUS: 'Status cannot exceed 100 characters',
  MAX_TOTAL: 'Total cannot exceed 10 characters',
  MAX_VIEW: 'View cannot exceed 10 characters',
  MAX_SALES: 'Sales cannot exceed 10 characters',
  MAX_CONVERSION: 'Conversion cannot exceed 5 characters',
};

export { EMPTY_TEXT, MESSAGE };
