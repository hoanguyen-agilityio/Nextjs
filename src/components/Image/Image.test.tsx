import { render } from '@testing-library/react';
import ImageCustom from '.';
import { IMAGE } from '@/constants';

describe('Image component', () => {
  test('Image Component With Displayed Image', () => {
    const { container } = render(
      <ImageCustom
        src={IMAGE.DEFAULT}
        width={200}
        height={300}
        alt="Image default"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test('Image Component With Displayed fallback image', () => {
    const { container } = render(
      <ImageCustom width={200} height={300} alt="fallback image" />,
    );
    expect(container).toMatchSnapshot();
  });
});
