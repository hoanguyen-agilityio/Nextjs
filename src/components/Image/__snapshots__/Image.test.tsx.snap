// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component Image Component With Displayed Image 1`] = `
<div>
  <div
    class="relative shadow-black/5 shadow-none rounded-large"
    style="background-image: url(https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSnOyW9pu8TBhOqw5tUQh_5A6lrIYfmpu4c3GkV7t6pcfu7oTILl9FAm2nX8n45uLHdSpk&usqp=CAU); max-width: 200px;"
  >
     
    <img
      alt="Image default"
      class="relative z-10 opacity-0 shadow-black/5 data-[loaded=true]:opacity-100 shadow-none transition-transform-opacity motion-reduce:transition-none !duration-300 rounded-large"
      height="300"
      loading="lazy"
      src="https://picsum.photos/200/300"
      style="height: 300px;"
      width="200"
    />
  </div>
</div>
`;

exports[`Image component Image Component With Displayed fallback image 1`] = `
<div>
  <div
    class="relative shadow-black/5 shadow-none rounded-large"
    style="background-image: url(https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSnOyW9pu8TBhOqw5tUQh_5A6lrIYfmpu4c3GkV7t6pcfu7oTILl9FAm2nX8n45uLHdSpk&usqp=CAU); max-width: 200px;"
  >
     
    <img
      alt="fallback image"
      class="relative z-10 opacity-0 shadow-black/5 data-[loaded=true]:opacity-100 shadow-none transition-transform-opacity motion-reduce:transition-none !duration-300 rounded-large"
      height="300"
      loading="lazy"
      style="height: 300px;"
      width="200"
    />
  </div>
</div>
`;
